src/gui/newwizard/setupwizardcontroller.cpp

```diff
@@ -101,8 +101,10 @@ void SetupWizardController::nextStep(std::optional<PageIndex> currentPage, std::
 
                 // fix scheme if necessary
                 // using HTTPS as a default is a really good idea nowadays, users can still enter http:// explicitly if they wish to
-                if (QUrl(userProvidedUrl).isRelative() && !userProvidedUrl.isEmpty()) {
-                    userProvidedUrl.prepend(QStringLiteral("https://"));
+                if (!userProvidedUrl.startsWith(QStringLiteral("http://")) && !userProvidedUrl.startsWith(QStringLiteral("https://"))) {
+                    static const QString defaultScheme(QStringLiteral("https://"));
```

Comment:
Move to anonymous namespace.

Remove the qDebug/promise to implement a logging category in an upcoming pr.

What was wrong with https://github.com/owncloud/client/blob/ca7f788b1ca68ad3ea8062c3f3e537793df89b93/src/gui/wizard/owncloudsetuppage.cpp#L123

now you will convert `foo://bar` to `https://foo://bar`.
Does Qt detect `foo:8080` wrong?